core/modules/book/book.admin.inc

```diff
@@ -218,6 +217,9 @@ function book_admin_edit_submit($form, &$form_state) {
         $node->save();
         watchdog('content', 'book: updated %title.', array('%title' => $node->title), WATCHDOG_NOTICE, l(t('view'), 'node/' . $node->nid));
       }
+
+      // Assume it's been updated and clear the navigation cache for the item.
+      _book_cache_clear($row['#item']['bid'], TRUE);
```

Comment:
This invocation should be outside the loop. Otherwise we are clearing the cache on every single row of the book outline.